memo/src/instruction.rs

```diff
@@ -0,0 +1,49 @@
+use crate::error::MemoError;
+use solana_sdk::{entrypoint::ProgramResult, info, program_error::ProgramError};
+use std::{mem::size_of, str::from_utf8};
+
+/// Instructions supported by the memo program
+#[repr(C)]
+#[derive(Clone, Debug, PartialEq)]
+pub enum Instruction<'a> {
+    /// Validate UTF-8 encoded characters
+    Utf8(&'a str),
```

Comment:
Can you follow the verb-subject naming convention that the other programs use?  Maybe `AddMemo`?

Reply:
`AddUtf8` or `AddUtf8Memo`? The whole point of having different variants is the encoding type

Reply:
Bumping the program id would be a nice mechanism for that too

Reply:
Otherwise we're asking all non-Rust users to append 8 zero bytes in front of their UTF-8 string right?  That's just weird IMO

Reply:
How about `AddUtf8String` so that it reads `memo_instruction::AddUtf8String` without duplicating the word "memo".

Reply:
It's just one byte, but yeah